script/testScripts/gov/AddRewardsToNitroPool/ProposeAddRewardsToNitroPool.s.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.19;
+
+import {JSONScript} from '@script/testScripts/gov/JSONScript.s.sol';
+import {ODGovernor} from '@contracts/gov/ODGovernor.sol';
+import {IERC20} from '@openzeppelin/token/ERC20/IERC20.sol';
+
+// Mock contract for NitroPool that will be used in the script. This is used to be very clear about what we are proposing.
+interface NitroPool {
+  function addRewards(uint256 amountToken1, uint256 amountToken2) external;
+}
+
+/// @title ProposeAddRewardsToNitroPool Script
+/// @author OpenDollar
+/// @notice Script to propose adding rewards to NitroPool via ODGovernance
+/// @dev NOTE This script requires the following env vars in the REQUIRED ENV VARS section below
+/// @dev This script is used to propose adding rewards to NitroPool, we first approve ERC20 to NitroPool and then call addRewards
+/// @dev The script will output a JSON file with the proposal data to be used by the QueueProposal and ExecuteProposal scripts
+/// @dev In the root, run: export FOUNDRY_PROFILE=governance && forge script --rpc-url <RPC_URL> script/testScripts/gov/AddRewardsToNitroPool/ProposeAddRewardsToNitroPool.s.sol
+contract ProposeAddRewardsToNitroPool is JSONScript {
+  function run() public {
+    /// REQUIRED ENV VARS ///
+    address governanceAddress = vm.envAddress('GOVERNANCE_ADDRESS');
+    address nitroPoolAddress = vm.envAddress('NITRO_POOL_ADDRESS');
+    address rewardTokenAddress1 = vm.envAddress('REWARD_TOKEN_ADDRESS_1');
+    address rewardTokenAddress2 = vm.envAddress('REWARD_TOKEN_ADDRESS_2');
+    uint256 amountTokenReward1 = vm.envUint('AMOUNT_TOKEN_REWARD_1');
+    uint256 amountTokenReward2 = vm.envUint('AMOUNT_TOKEN_REWARD_2');
+
+    /// END REQUIRED ENV VARS ///
+
+    ODGovernor gov = ODGovernor(payable(governanceAddress));
+    address[] memory targets = new address[](1);
+    {
+      targets[0] = rewardTokenAddress1;
+      targets[1] = rewardTokenAddress2;
+      targets[2] = nitroPoolAddress;
+    }
+    // No values needed
+    uint256[] memory values = new uint256[](1);
+    {
+      values[0] = 0;
+      values[1] = 0;
+      values[2] = 0;
+    }
+
+    // Get calldata for:
+    //  - ERC20 Approve Nitro Pool (Token1)
+    //  - ERC20 Approve Nitro Pool (Token2)
+    //  - NitroPool.addRewards
+    bytes[] memory calldatas = new bytes[](1);
+    calldatas[0] = abi.encodeWithSelector(IERC20.approve.selector, nitroPoolAddress, amountTokenReward1);
+    calldatas[1] = abi.encodeWithSelector(IERC20.approve.selector, nitroPoolAddress, amountTokenReward2);
```

Comment:
Can you create it so that it only calls `approve` and `addRewards` accordingly depending on the number of tokens (eg if `amountTokenReward2` is non-zero, the extra approval is omitted)? 

If we ran this to add only ODG rewards, then there would be an extra approval call, which could be confusing

Reply:
done, now the proposal will construct the right calldatas/targets based on tokens amount.

Also I hit `stack too deep`, that's why I start using the struct as a placeholder.